vignettes/deepdive.Rmd

```diff
@@ -0,0 +1,404 @@
+
+## What is `{xportr}` validating in these Data Packages?
+
+The `xpt` Version 5 files form the backbone of any successful Submission and are govern by quite a lot of rules and suggested guidelines. As you are preparing your data packages for submission the suite of core `{xportr}` functions, plus `xportr_write()`, helps to check that your datasets are submission compliant. The package checks many of the latest rules laid out in the [Study Data Technical Conformance Guide](https://www.fda.gov/regulatory-information/search-fda-guidance-documents/study-data-technical-conformance-guide-technical-specifications-document), but please note that it is not yet an exhaustive list of checks. We envision that users are also submitting their `xpts` and metadata to additional validation software.
+
+Each of the core `{xportr}` functions for applying labels, types, formats, order and lengths provides feedback to users on submission compliance. However, a final check is implemented when `xportr_write()` is called to create the `xpt`. `xportr_write()` calls [`xpt_validate()`](https://github.com/atorus-research/xportr/blob/231e959b84aa0f1e71113c85332de33a827e650a/R/utils-xportr.R#L174), which is a behind the scenes/non-exported function that does a final check for compliance.  At the time of `{xportr} v0.3.0` we are checking the following when a user writes out an `xpt` file.:
+
+<img src="xpt_validate.png" alt="validate" style="width:800px;"/>
+
+
+# {xportr} in action
+
+In this section, we are going to explore the 5 core `{xportr}` functions using: 
+
+* An ADSL ADaM dataset from the Pilot 3 Submission to the FDA
+* The ADSL ADaM Specification File from the Pilot 3 Submission to the FDA
+
+We will focus on warning and error messaging with contrived examples from these functions by manipulating either the datasets or the specification files.
+
+**NOTE:** We have made the ADSL and Spec available in this package. Users can find additionl datasets and specification files on our [repo](https://github.com/atorus-research/xportr) in the `example_data_specs` folder. This is to keep the package to a minimum size.  
+
+
+## Using `options()` and `xportr_metadata()` to enhance your experience.
+
+Before we dive into the functions, we want to point out some quality of life utilities to make your `xpt` generation life a little bit easier.
+
+* `options()` 
+* `xportr_metadata()` 
+
+**NOTE:** As long as you have a well-defined _metadata object_ you do NOT need to use `options()` or `xportr_metadata()`, but we find these handy to use and think they deserve a quick mention!
+
+## You got `options()`
```

Comment:
I think we can also have a small section after the cleverly named _You got options_ like _verbose your choices_ to explain what this does because just like options verbose is something that many functions get affected so it's worth mentioning it before diving into them. WDYT?

Reply:
haha - how about _"Are we being too verbose?"_

Reply:
:D yessss

Reply:
Added a small section